src/pages/examples/DrawingToolsPage.tsx

```diff
@@ -0,0 +1,283 @@
+
+	useEffect(() => {
+		const drawingToolsApi = useDrawingToolsApiStore.getState().drawingToolsApi;
+		let planeRestrictionApi: PlaneRestrictionApi | undefined;
+		if(drawingToolsApi)
+			planeRestrictionApi = Object.values(drawingToolsApi.restrictions).find(restriction => restriction instanceof PlaneRestrictionApi) as PlaneRestrictionApi | undefined;
+
+		setMaterialParameters(
+			[
+				{
+					definition: {
+						id: PARAMETER_NAMES.START_DRAWING,
+						name: "Start Drawing",
+						defval: (drawingToolsApi !== undefined) + "",
+						type: ShapeDiverResponseParameterType.BOOL,
+						hidden: false
+					},
+				},
+				{
+					definition: {
+						id: PARAMETER_NAMES.SHOW_POINT_LABELS,
+						name: "Show Point Labels",
+						defval: (drawingToolsApi?.showPointLabels ?? false) + "",
+						type: ShapeDiverResponseParameterType.BOOL,
+						hidden: !outputNameDrawingTools
+					}
+				},
+				{
+					definition: {
+						id: PARAMETER_NAMES.SHOW_DISTANCE_LABELS,
+						name: "Show Distance Labels",
+						defval: (drawingToolsApi?.showDistanceLabels ?? false) + "",
+						type: ShapeDiverResponseParameterType.BOOL,
+						hidden: !outputNameDrawingTools
+					}
+				},
+				{
+					definition: {
+						id: PARAMETER_NAMES.GRID_SNAPPING,
+						name: "Grid Snapping",
+						defval: (planeRestrictionApi?.gridRestrictionApi.enabled ?? false) + "",
+						type: ShapeDiverResponseParameterType.BOOL,
+						hidden: !outputNameDrawingTools
+					}
+				},
+				{
+					definition: {
+						id: PARAMETER_NAMES.GRID_UNIT,
+						name: "Grid Unit",
+						defval: (planeRestrictionApi?.gridRestrictionApi.gridUnit ?? 1) + "",
+						min: 1,
+						max: 10,
+						type: ShapeDiverResponseParameterType.INT,
+						hidden: !outputNameDrawingTools
+					}
+				},
+				{
+					definition: {
+						id: PARAMETER_NAMES.ANGULAR_SNAPPING,
+						name: "Angular Snapping",
+						defval: (planeRestrictionApi?.angularRestrictionApi.enabled ?? false) + "",
+						type: ShapeDiverResponseParameterType.BOOL,
+						hidden: !outputNameDrawingTools
+					}
+				},
+				{
+					definition: {
+						id: PARAMETER_NAMES.ANGULAR_SECTIONS,
+						name: "Angular Sections",
+						defval: (planeRestrictionApi?.angularRestrictionApi.angleStep ?? 8 / Math.PI) * Math.PI + "",
+						min: 2,
+						max: 24,
+						type: ShapeDiverResponseParameterType.INT,
+						hidden: !outputNameDrawingTools
+					}
+				}
+			]
+		);
+	}, [drawingToolsApi, setDrawingToolsApi, outputNameDrawingTools]);
+
+
+	// define the custom drawing tools parameters and a handler for changes
+	const customSessionId = "mysession";
+	useDefineGenericParameters(customSessionId, false /* acceptRejectMode */,
+		materialParameters,
+		async (values) => {
+			if (PARAMETER_NAMES.START_DRAWING in values)
+				setOutputNameDrawingTools(""+values[PARAMETER_NAMES.START_DRAWING] === "true" ? "DrawingToolsOptions" : "");
+
+			const drawingToolsApi = useDrawingToolsApiStore.getState().drawingToolsApi;
```

Comment:
Once we return the drawingToolsApi from the useDrawingTools hook, you can also use a useRef to access it here.